docs/reference/contributing/guidelines/license.md

```diff
@@ -106,6 +106,8 @@ If you decide to use a different license for your work, follow the same pattern:
 
 #### Mbed OS principles
 
-Mbed OS uses these same basic principles for its source code and library distributions. So source code we own is distributed under the Apache 2.0 license, and binary blobs are released under the Permissive Binary License. Software parts from third parties that were already licensed under a different license are available under that original license.
+By creating a pull request on GitHub, you are agreeing to license your contributions under the same license as the original code. Note that Mbed software is released under a permissive license for all new Mbed contributions. Arm will not accept proprietary code or any code under a "copyleft" license. Accepting such code would prevent us from continuing to distribute our code under the permissive license and stall adoption of our code by third parties.
```

Comment:
This might be way too nitpicky a question, but do we need to specify a pull request for Mbed OS? If we're already being this specific and using legalese...

Reply:
What's your proposal?

Reply:
By creating a pull request to Mbed OS on GitHub...

Reply:
Lemme check on that preposition...

Reply:
Kevin B says it's 'against', so I'll go with that. One moment.